sympy/polys/rings.py

```diff
@@ -3005,3 +3178,295 @@ def sqf_list(f, all=False):
 
     def factor_list(f):
         return f.ring.dmp_factor_list(f)
+
+def cont_prim(p, x):
+    """Returns the content and primitive part of a polynomial with respect to a
+    specified variable."""
+
+    coeffs = p.coeff_split({x})
+    cont = gcd_prs(coeffs)
+    prim = p.exquo(cont)
+    return cont, prim
+
+def monomial_extract(polynomials):
+    """
+    Extracts any common monomial from the polynomials.
+
+    Examples
+    ========
+
+    >>> from sympy.polys.rings import monomial_extract
+    >>> from sympy import ZZ, ring
+    >>> R, x, y = ring("x, y", ZZ)
+
+    >>> f = x + y
+    >>> g = x**2*y + x*y
+    >>> polynomials = [f, g]
+    >>> monomial_extract(polynomials)
+    ([x + y, x**2*y + x*y], 1)
+
+    >>> f = x**2*y
+    >>> g = x**2*y + x*y
+    >>> polynomials = [f, g]
+    >>> monomial_extract(polynomials)
+    ([x, x + 1], x*y)
+
+    """
+    ring = polynomials[0].ring
+    domain = ring.domain
+
+    zero_monom = ring.zero_monom
+    monoms = chain(*polynomials)
+
+    # Check for the presence of zero_monom in any polynomial
+    if any(zero_monom in poly for poly in polynomials):
+        return polynomials, domain.one
+
+    monom_gcd = tuple(map(min, zip(*monoms)))
+
+    if monom_gcd == ring.zero_monom:
+        return polynomials, domain.one
+    else:
+        d = ring({monom_gcd: ring.domain.one})
+        p = [pi.exquo(d) for pi in polynomials]  # TODO: Use monomial_ldiv
+        return p, d
+
+def gcd_coeffs(coeff_lst, domain):
+    """
+    Return the greatest common divisor (GCD) of a list of coefficients of a
+    polynomial.
```

Comment:
This is really just `ngcd` or `gcdn`. There is no reason to call this `coeffs` because nothing about it is specific to polynomials or coefficients.

Probably this would be best as a `Domain` method.